Automattic/wp-calypso
# Store: Add withWooCommerceSite HOC

PUtting on hold in favor of #22961


Note: The ethos of this PR is explained in the README in #21274

This PR tries out a `withWooCommerceSite` HOC and some example code to use it.
It modifies `ProductForm` to use the HOC as a demonstration of how it
works.

To Test:
1. Edit an existing product. Everything should look/behave the same.
2. Add a new product. Everything should look/behave the same.
3. In Redux DevTools, look at state. Under `extensions.woocommerce.wcApi[ <site id> ].endpoints.products`, these products should be included in a list of products by id.
